src/rand_impls.rs

```diff
@@ -247,6 +366,64 @@ impl<T:Rand> Rand for Option<T> {
     }
 }
 
+impl<T: Rand, E: Rand> Rand for Result<T, E> {
+    #[inline]
+    fn rand<R: Rng>(rng: &mut R) -> Result<T, E> {
+        if rng.gen() {
+            Ok(rng.gen())
+        } else {
+            Err(rng.gen())
+        }
+    }
+}
+
+impl<T: Rand> Rand for Box<T> {
+    #[inline]
+    fn rand<R: Rng>(rng: &mut R) -> Box<T> {
+        Box::new(rng.gen())
+    }
+}
+
+impl<T: Rand> Rand for Rc<T> {
+    #[inline]
+    fn rand<R: Rng>(rng: &mut R) -> Rc<T> {
+        Rc::new(rng.gen())
+    }
+}
+
+impl<T: Rand> Rand for Arc<T> {
+    #[inline]
+    fn rand<R: Rng>(rng: &mut R) -> Arc<T> {
+        Arc::new(rng.gen())
+    }
+}
+
+// TODO: remove Copy
+impl<T: Copy + Rand> Rand for Cell<T> {
```

Comment:
I doubt you'll ever be able to remove Copy. So remove the TODO?

Reply:
Actually, doing so only requires a minimum rustc bump.